### ISSUE TITLE
`java.lang.OutOfMemoryError` when fetching getAllpaths

### ISSUE THREAD

**Issue by jxnu-liguobin** [jgrapht/jgrapht#1161]
```
 * JGraphT version:
 * Java version (java -version)/platform:  
```

**Issue**

We have 400,000 nodes, and when I use getAllPaths(String source, String target) method (I set the depth to 100.) to get all paths from each node to the target node, the program always gets `java.lang.OutOfMemoryError`, Each node may have more than 100 edges.

**Steps to reproduce (small coding example)**

No

**Expected behaviour**

getAllPaths method can use Iterator instead of List ? or stream?

**Other information**

**Comment by jsichi** [comment]
Right, the current implementation does not provide an iterator, and the number of paths can explode, so it's prone to exhausting memory.

There are iterators available for the k-shortest path implementations.

**Comment by jxnu-liguobin** [comment]
> Right, the current implementation does not provide an iterator, and the number of paths can explode, so it's prone to exhausting memory.
> 
> 

Thank you, I'll try it.
